engine/src/main/java/com/redhat/cloud/notifications/health/LivenessService.java

```diff
@@ -0,0 +1,44 @@
+package com.redhat.cloud.notifications.health;
+
+import io.smallrye.health.api.AsyncHealthCheck;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Liveness;
+import org.hibernate.reactive.mutiny.Mutiny;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@Liveness
+@ApplicationScoped
+public class LivenessService implements AsyncHealthCheck {
+
+    @Inject
+    Mutiny.SessionFactory sessionFactory;
+
+    @Inject
+    KafkaConsumedTotalChecker kafkaConsumedTotalChecker;
+
+    @Override
+    public Uni<HealthCheckResponse> call() {
+        HealthCheckResponseBuilder response = HealthCheckResponse.named("Notifications readiness check");
+        if (kafkaConsumedTotalChecker.isDown()) {
```

Comment:
This assumes that we are always sure that at least 1 engine is processing kafka messages - is that true?

Reply:
Not exactly. This is the exact same thing as [this current backend check](https://github.com/RedHatInsights/notifications-backend/blob/aa7369e12091e053f50dd0146aa94be8710555d7/backend/src/main/java/com/redhat/cloud/notifications/health/LivenessService.java#L34-L38). This class does not introduce anything new but replicates most of the current `backend` behavior into the `engine`.

Reply:
To be more precise, if any `engine` fails to proceed Kafka messages for a while (delay is configuration-dependent) then it will be marked as unhealthy and restarted.